libs/community/langchain_community/vectorstores/pgvector.py

```diff
@@ -508,65 +508,61 @@ def _results_to_docs_and_scores(self, results: Any) -> List[Tuple[Document, floa
-            filter_by_metadata = self.EmbeddingStore.cmetadata[key].astext.not_in(
-                value_case_insensitive[NIN]
-            )
-        elif BETWEEN in map(str.lower, value):
-            filter_by_metadata = self.EmbeddingStore.cmetadata[key].astext.between(
-                str(value_case_insensitive[BETWEEN][0]),
-                str(value_case_insensitive[BETWEEN][1]),
-            )
-        elif GT in map(str.lower, value):
-            filter_by_metadata = self.EmbeddingStore.cmetadata[key].astext > str(
-                value_case_insensitive[GT]
-            )
-        elif LT in map(str.lower, value):
-            filter_by_metadata = self.EmbeddingStore.cmetadata[key].astext < str(
-                value_case_insensitive[LT]
-            )
-        elif NE in map(str.lower, value):
-            filter_by_metadata = self.EmbeddingStore.cmetadata[key].astext != str(
-                value_case_insensitive[NE]
-            )
-        elif EQ in map(str.lower, value):
-            filter_by_metadata = self.EmbeddingStore.cmetadata[key].astext == str(
-                value_case_insensitive[EQ]
-            )
-        elif LIKE in map(str.lower, value):
-            filter_by_metadata = self.EmbeddingStore.cmetadata[key].astext.like(
-                value_case_insensitive[LIKE]
-            )
-        elif CONTAINS in map(str.lower, value):
-            filter_by_metadata = self.EmbeddingStore.cmetadata[key].astext.contains(
-                value_case_insensitive[CONTAINS]
-            )
-        elif OR in map(str.lower, value):
-            or_clauses = [
-                self._create_filter_clause(key, sub_value)
-                for sub_value in value_case_insensitive[OR]
-            ]
-            filter_by_metadata = sqlalchemy.or_(*or_clauses)
-        elif AND in map(str.lower, value):
-            and_clauses = [
-                self._create_filter_clause(key, sub_value)
-                for sub_value in value_case_insensitive[AND]
-            ]
-            filter_by_metadata = sqlalchemy.and_(*and_clauses)
-
-        else:
-            filter_by_metadata = None
-
-        return filter_by_metadata
+    def _create_filter_clause(self, filters):  # type: ignore[no-untyped-def]
+        from langchain.chains.query_constructor.ir import Comparator, Operator
+
+        COMPARISONS = {
+            Comparator.EQ: lambda f, v: f == str(v),
+            Comparator.NE: lambda f, v: f != str(v),
+            Comparator.LT: lambda f, v: f < str(v),
+            Comparator.LTE: lambda f, v: f <= str(v),
+            Comparator.GT: lambda f, v: f > str(v),
+            Comparator.GTE: lambda f, v: f >= str(v),
+            Comparator.LIKE: lambda f, v: f.like(v),
+            Comparator.IN: lambda f, v: f.in_(v),
+            Comparator.NIN: lambda f, v: f.not_in(v),
+            Comparator.CONTAIN: lambda f, v: f.contains(v),
+            "contains": lambda f, v: f.contains(v),
```

Comment:
cc @baskaryan looks like we've got some operations not covered by the existing enums?